router/http_test.go

```diff
@@ -478,23 +489,25 @@ func (s *S) TestStickyHTTPRouteWebsocket(c *C) {
 	}
 }
 
-func (s *S) TestNoStickyHeaderAtBackend(c *C) {
-	h := http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
-		_, ok := req.Header[hdrUseStickySessions]
-		c.Assert(ok, Equals, false)
-	})
-
-	srv := httptest.NewServer(h)
+func (s *S) TestFailedWebSocketClosed(c *C) {
+	srv := httptest.NewServer(http.HandlerFunc(http.NotFound))
 	defer srv.Close()
 
 	l := newHTTPListener(c)
+	url := "http://" + l.Addr
 	defer l.Close()
 
 	addHTTPRoute(c, l)
-
 	discoverdRegisterHTTP(c, l, srv.Listener.Addr().String())
 
-	assertGet(c, "http://"+l.Addr, "example.com", "")
+	req := newReq(url, "example.com")
+	req.Header.Set("Connection", "Upgrade")
+	req.Header.Set("Upgrade", "websocket")
+	res, err := httpClient.Do(req)
+	defer res.Body.Close()
+
+	c.Assert(err, IsNil)
+	c.Assert(res.Close, Equals, true)
```

Comment:
Turns out this behavior wasn't correct after all. See 616d112a9fd1e6d195239705cdf98699c5792393 for an explanation and my suggested fix.